### DISCUSSION TITLE
Is there easy way to implement dynamic file roots based on url?

### DISCUSSION THREAD

**Discussion by dejanmilosevic0** [coder/code-server#2661] [Ideas]
If url is in conflict with visual code, query parameter is ok as well, i just need to have multiple urls like:

vscode/?root=project1
vscode/?root=project2

and before digging into code i just wanted to asked if maybe someone tried this feature before me.

**Comment by code-asher** [comment]
You can specify paths with `folder` or workspace files with `workspace`.

So for example:

```
vscode/?folder=/path/to/folder
vscode/?workspace=/path/to/file.workspace
```

**Reply by dejanmilosevic0** [reply]
If i go to:

https://localhost:8443/?folder=/home/test

I get:

Cannot GET /

**Reply by code-asher** [reply]
How are you running code-server? The default port is 8080, maybe that's why?

**Reply by code-asher** [reply]
Could also be related to TLS since you're using https. Are you using `--cert`?

**Reply by dejanmilosevic0** [reply]
I managed to get it working, but I am just wondering user still can go to file open and access other directories, is there a way to lock user on directory based on a dynamic url?

**Reply by code-asher** [reply]
Nope, but we have an open issue for implementing that: https://github.com/cdr/code-server/issues/1834
